sql-plugin/src/main/scala/com/nvidia/spark/rapids/ColumnarOutputWriter.scala

```diff
@@ -165,12 +170,14 @@ abstract class ColumnarOutputWriter(context: TaskAttemptContext,
       }
       val startTimestamp = System.nanoTime
       withResource(sb.getColumnarBatch()) { cb =>
-        RmmRapidsRetryIterator.withRestoreOnRetry(cr) {
-          withResource(new NvtxRange(s"GPU $rangeName write", NvtxColor.BLUE)) { _ =>
-            withResource(GpuColumnVector.from(cb)) { table =>
-              scanTableBeforeWrite(table)
-              anythingWritten = true
-              tableWriter.write(table)
+        withResource(transform(cb)) { transformed =>
```

Comment:
would be good to add a TODO + follow on issue to clean this up. If I understand correctly, if we keep the `Table` around we won't need to `transform`.

Reply:
Just to clarify this, we do still need to perform the transform. Previously, it was performed before creating the spillable batch, and now it is performed after getting the batch from the spillable batch. If we had a `SpillableTable` we could go back to performing it earlier.